test/components/views/dialogs/DevtoolsDialog-test.tsx

```diff
@@ -0,0 +1,67 @@
+/*
+Copyright 2023 The Matrix.org Foundation C.I.C.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import React from "react";
+import { getByLabelText, render } from "@testing-library/react";
+import { Room } from "matrix-js-sdk/src/models/room";
+import { MatrixClient } from "matrix-js-sdk/src/client";
+import userEvent from "@testing-library/user-event";
+import { sleep } from "matrix-js-sdk/src/utils";
+
+import { stubClient } from "../../../test-utils";
+import { MatrixClientPeg } from "../../../../src/MatrixClientPeg";
+import MatrixClientContext from "../../../../src/contexts/MatrixClientContext";
+import DevtoolsDialog from "../../../../src/components/views/dialogs/DevtoolsDialog";
+
+describe("DevtoolsDialog", () => {
+    let cli: MatrixClient;
+    let room: Room;
+
+    function getComponent(roomId: string, onFinished = () => true) {
+        return render(
+            <MatrixClientContext.Provider value={cli}>
+                <DevtoolsDialog roomId={roomId} onFinished={onFinished} />
+            </MatrixClientContext.Provider>,
+        );
+    }
+
+    beforeEach(() => {
+        stubClient();
+        cli = MatrixClientPeg.get();
+        room = new Room("!id", cli, "@alice:matrix.org");
+
+        jest.spyOn(cli, "getRoom").mockReturnValue(room);
+    });
+
+    it("renders the devtools dialog", () => {
+        const { asFragment } = getComponent(room.roomId);
+        expect(asFragment()).toMatchSnapshot();
+    });
+
+    it("copies the roomid", async () => {
+        const user = userEvent.setup();
+        const { container } = getComponent(room.roomId);
+
+        const copyBtn = getByLabelText(container, "Copy");
+        user.click(copyBtn);
+
+        // Not great, but I could not find another way to wait for the clipboard
+        // to be populated
+        // Ideally would prefer to write `waitFor(() => expect(navigator.clipboard.write).toHaveBeenCalled())
```

Comment:
This test appears to be failing (at least according to what I can see on Github, so I wonder if this idea was good, and it was the next test that was failing?

Alternatively I wonder in this case if you could turn `navigator.clipboard` into a mock fn by spying on it, then check to see that it has been called and what it has been called with?

Reply:
`userEvent.setup()` is supposed to do just that... https://testing-library.com/docs/user-event/clipboard/

But it does not seem to work here, I'm failing to find much information online